spec/filters/kv_spec.rb

```diff
@@ -1147,11 +1144,21 @@
       plugin.filter(event)
     end
     context 'when a custom tag is defined' do
-      let(:options) { super().merge("tag_on_failure" => "KV-ERROR")}
+      let(:options) { super().merge("tag_on_failure" => ["KV-ERROR"])}
       it 'tags the event with the custom tag' do
         plugin.filter(event)
         expect(event.get('tags')).to_not be_nil
         expect(event.get('tags')).to include('KV-ERROR')
+        expect(event.get('tags')).to_not include('_kv_filter_error')
+      end
+    end
+    context 'when multiple custom tags are defined' do
+      let(:options) { super().merge("tag_on_failure" => ["_kv_filter_error", "_kv_filter_pattern"])}
+      it 'tags the event with the custom tag' do
+        plugin.filter(event)
+        expect(event.get('tags')).to_not be_nil
+        expect(event.get('tags')).to include('_kv_filter_error')
+        expect(event.get('tags')).to include('_kv_filter_pattern')
```

Comment:
Lets include two tags that do _not_ overlap the default, so that we can specify that the result doesn't include the default when an array is presented:

```suggestion
      let(:options) { super().merge("tag_on_failure" => ["kv_FAIL_one", "_kv_fail_TWO"])}
      it 'tags the event with the custom tags' do
        plugin.filter(event)
        expect(event.get('tags')).to_not be_nil
        expect(event.get('tags')).to include('kv_FAIL_one')
        expect(event.get('tags')).to include('_kv_fail_TWO')
        expect(event.get('tags')).to_not include('_kv_filter_error')
```